front/.dependency-cruiser.js

```diff
@@ -13,6 +13,30 @@ module.exports = {
         pathNot: 'app/modules/index.ts',
       },
     },
+    {
+      name: 'no-import-to-component-library',
+      severity: 'error',
+      comment:
+        'Direct imports to the component library from the rest of the codebase are not allowed',
+      from: {
+        path: 'app/component-library/',
+      },
+      to: {
+        path: 'app/',
+        pathNot: 'app/component-library/',
+      },
+    },
+    {
+      name: 'no-imports-from-component-library-without-using-an-alias',
+      comment: 'Imports from the component library should use an alias',
+      severity: 'error',
+      from: { pathNot: 'app/component-library/' },
+      to: {
+        path: 'app/component-library/',
+        dependencyTypesNot: ['aliased'],
+      },
+    },
```

Comment:
I can import from 'component-library' without getting an error:

`import { Thead, Tr, Th, Checkbox, colors } from 'component-library';
`

But maybe we don't need this rule? What would be the value of it?

Reply:
We do need it but I ran into a limitation of our dependency cruiser version. I need to update it to solve it but for that - we need to update node as it doesn't support v16 anymore. So I think we should leave it as is for now and once we upgrade our node version, we can change this, too.

It's still useful as it at least disallows relative imports.

Reply:
Not completely following, but seems like you know what you're doing. So let's maybe add it as a todo to the ticket? Good to be merged then I guess. ✅

Reply:
I'll make sure I document it, yes :)